fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java

```diff
@@ -226,14 +255,44 @@ protected Plan rewriteQueryByView(MatchMode matchMode,
     }
 
     // only support sum roll up, support other agg functions later.
-    private AggregateFunction rollup(AggregateFunction originFunction,
-            Expression mappedExpression) {
-        Class<? extends AggregateFunction> rollupAggregateFunction = originFunction.getRollup();
+    private Function rollup(AggregateFunction queryFunction,
+            Expression queryFunctionShuttled,
+            Map<Expression, Expression> mvExprToMvScanExprQueryBased) {
+        Expression rollupParam = null;
+        if (mvExprToMvScanExprQueryBased.containsKey(queryFunctionShuttled)) {
+            // function can not rewrite by view
+            rollupParam = mvExprToMvScanExprQueryBased.get(queryFunctionShuttled);
+        } else {
+            // try to use complex roll up param
+            // eg: query is count(distinct param), mv sql is bitmap_union(to_bitmap(param))
+            for (Expression mvExprShuttled : mvExprToMvScanExprQueryBased.keySet()) {
+                if (!(mvExprShuttled instanceof Function)) {
+                    continue;
+                }
+                if (isAggregateFunctionEquivalent(queryFunction, (Function) mvExprShuttled)) {
+                    rollupParam = mvExprToMvScanExprQueryBased.get(mvExprShuttled);
+                }
+            }
+        }
+        if (rollupParam == null) {
+            return null;
+        }
+        // do roll up
+        Class<? extends Function> rollupAggregateFunction = queryFunction.getRollup();
         if (rollupAggregateFunction == null) {
             return null;
         }
         if (Sum.class.isAssignableFrom(rollupAggregateFunction)) {
-            return new Sum(originFunction.isDistinct(), mappedExpression);
+            return new Sum(queryFunction.isDistinct(), rollupParam);
+        }
+        if (Max.class.isAssignableFrom(rollupAggregateFunction)) {
+            return new Max(queryFunction.isDistinct(), rollupParam);
+        }
+        if (Min.class.isAssignableFrom(rollupAggregateFunction)) {
+            return new Min(queryFunction.isDistinct(), rollupParam);
+        }
+        if (BitmapUnionCount.class.isAssignableFrom(rollupAggregateFunction)) {
+            return new BitmapUnionCount(rollupParam);
```

Comment:
i think u should add a new trait named CouldRollUp and a new interface into this trait, such as `constructRollUp`, and return rollup object directly

Reply:
it's reasonable, i will fix it